### DISCUSSION TITLE
Is Microsoft really committed to MAUI?

### DISCUSSION THREAD

**Discussion by tschramme86** [dotnet/maui#10596] [General] [answered]
I do appreciate all the community effort here - but in the end .net MAUI is advertised by Microsoft and so perceived as mainly driven by Microsoft. And when such a big company announces such a framework as the new unification for UI stuff on all platforms, this comes with some expectations from your developer community.

Being one of these developers, it would be great to hear from Microsoft about their future plans for MAUI, if there is a commitment to this product and how they plan to fulfil this commitment.

Let's look at some facts, just based on the insights in this repo.
On the one hand...
- 1.1k out of 1.7k reported issues are bugs
- Over 100 issues are p/0 or p/1, so confirmed high severity + release critical bugs
- Over 200 issues with layout bugs - and yes, the layout engine seems to be really broken in some places, I tried with a bigger project
- Plus a lot of unreported bugs because they happen completely randomly (my app, for example, crashes about every 5th start when initializing the Shell - TargetInvocationException. Just start debugging again and it works. Mostly...)

On the other hand...
- A year ago there were between 50-100 commits per week, now there are roughly 15-35 per week and most of them are just dependency updates
- The number of open issues is almost static, no progress. No work on the issues, no transparent prioritisation
- Looks like Microsoft just want to get rid of those tickets when the MsftBot closes automatically tickets because of "inactivity" - where "inactivity" means that the bug was already confirmed in previous comments but the reporter just missed to answer some totally random standard questions within 3 days (this typical game to push tickets back to the customer...)

So comparing the number of issues with the development performance, Microsoft, how do you ever plan MAUI to get a stable product? And I'm just looking at this Repo here - don't want to start talking about all the bugs in the tools...

Ok, now you could say "why not continuing using Xamarin.Forms, the support has been extended recently". True, XF is more stable than MAUI. But also XF has still a lot of severe bugs, also in core components like the Shell. Known for over a year, no fixing. And what helps me an official support until 2024 when bugs are not fixed at all? There are 4 commits to XF in the last 3 months. XF is dead.

So what can we developers expect from Microsoft here in the next months?

**Comment by 3steve3** [comment]
> bugs

I've seen someone in a bug report say since this project is open source you can go and fix the bugs yourself. But it is backed by Microsoft and not some solo developer and it's shipped as stable. Maybe it'll get better in a year, although the current state is stable enough for me to make an Android and iOS project that's fairly small (about 10 views). 

I just get the feeling that the small Maui team is overwhelmed and Microsoft is refusing to increase funding for this project.

**Comment by gautambjain** [comment]
This question "Is Microsoft really committed to MAUI" has been ringing in my mind for last many weeks. Good that you asked.

I have been watching all the 3 repos - MAUI, WinUI and Windows App SDK. All the 3 seem to be in the same state. 

Microsoft, you have done a great job till now! We appreciate and support it (as of today). But these partly closed, partly opened source repos and your non-responsive, careless attitude sends out varied kinds of fears to the community.

**Comment by davidortinau** [comment, marked as answer]
I appreciate the question, and it's a fair one.

I could point you to https://dot.net/ where .NET MAUI headlines alongside Blazor, the recent .NET MAUI Focus event, or our public support policy https://dotnet.microsoft.com/en-us/platform/support/policy/maui. I could also reiterate statements of commitment.

However, what I think everyone really needs is to see the work. I hope that you'll continue with us in our pursuit of a product that demonstrates Microsoft's commitment. Here are some of the things we are doing:

* Our engineering roadmap for the next year is quality. You'll see very little new feature work.
* We are iterating on process changes such as the use of bots to triage, updates to how we triage, release, etc. This is documented here: https://github.com/dotnet/maui/tree/main/docs
* More engineering resources are focusing on resolving .NET MAUI issues

We are 5 months past release in May, and complete stable tooling is still in the works. I think a lot of commits early as compared to recently might be explained by a few factors. Early work was mostly porting code from Xamarin.Forms, repetitive work. The work now is troubleshooting and bug fixing which can take longer. The product is at a different phase of its maturing.

If anyone has a critical need in Xamarin.Forms, please let me know. We continue to service it while the bulk of our effort is on .NET MAUI.

How you can help:

* If you are impacted by a bug that is filed, add a reaction to the original description and add any details that might be missing. This will help us sort the issue during our triage and sprint planning
* File bugs with quality details. If we can suggest a workaround, we will until the issue can be addressed directly
* Consider contributing. It's Hacktoberfest after all. I find .NET MAUI easier to contribute to than Xamarin.Forms, and I have 2 PRs merged in the past few weeks. While a PR will take some time to make it into a service release, the same code can be used in your project until released, and then you can remove it. Join us on Discord for any help in contributing and navigating the source http://aka.ms/dotnet-discord

As always, anyone can reach me at david.ortinau@microsoft.com. I would love to hear about your struggles and your successes.

**Reply by tschramme86** [reply]
@davidortinau Many thanks for your immediate and very open response. I really appreciate and to be honest, such communication is what I (or we, if I consider the discussions here and may speak in the name of the community) missed in the past.

Fingers crossed for all of us that the effort of Microsoft in improving the quality will take effect soon.

And when there is some time in the next weeks I'll take your offer to contact you and share some successes. 
Currently I'm developing a bigger app for Android and soon iOS - over 25 pages + lots of custom controls, GPS tracking, Maps integration, Azure based cloud backend, Azure AD B2C based user accounts. Public release (at least as beta) planned for spring next year. Started in Xamarin Forms, now managed to upgrade to MAUI. I often asked myself if it wouldn't have been better to develop such an app with the native frameworks of the two platforms. But now, after spending that much time, there is no way back. And actually I managed to find at least (more or less good) workarounds for most of the flaws in MAUI, especially regarding the bugs in the layout engine. 
Happy to receive then your feedback on that.

**Reply by Majestic-Mustang** [reply]
@tschramme86 Is it possible to share your app with the community as sample (by stripping out code that can't be shared)? 
Sounds like we can learn a lot from your app. Thank you!

**Reply by Majestic-Mustang** [reply]
@davidortinau Thank you for the great reply!

The community strongly feels that Microsoft is skimping on the resources that MAUI needs.
MAUI has great potential and we want to see it succeed but the pace it's progressing isn't looking good. 
Hopefully Satya Nadella sees this and does something about it. :)

**Comment by developer9969** [comment]
@tschramme86  I could not have put any better - You put together in a constructive manner the sentiment that the all community feels at the moment. We are stuck with Xamarin and we would love to embrace Maui but we can't because of all the issues . 

@davidortinau  it must be a very difficult position that one with your hands tight and a ridiculous low budget - I can only assume. Otherwise your team would be three times the size.

I am sure you are aware and already seen this link but that is the state we are in.
https://visualstudiomagazine.com/articles/2022/09/29/net-maui-complaints.aspx?oly_enc_id=1272C3125745E9V

We all want Maui to be a huge success - but its tricky because of the state of the tooling and maui itself.

I guess we need to hear more from you at your stand up on what you actually doing with the fixes.

thanks

**Comment by kaz911** [comment]
I think, as a beginner on MAUI, that it is a bit frighting that even the provided examples and learning resources provided by Microsoft - have not been checked for cross platform compatibility. 

It feels a bit like the developers doing the learning materials - just focus on ONE platform while they develop the learning material - and do not give a HOOT about the other platforms. If the other platforms compile and work - it is fine - and if they compile and don't work - nothing is done about it. 

I have now been going through TONS of learning material both from Microsoft and 3rd parties (MVP's) - and all seem "MAUI" brainwashed - none of them even mention "This might not work on platform X/Y/Z due to bugs in MAUI" 

I have so far not been able to get a SINGLE learning project work equally well on all platforms. It a lot of simple things that does not work - like visual indication of switching buttons "on/off" - drawing commands - formatting -  and many other "very basic" items. So it might work on Mac and iOS and then not on Android and Windows. Or it might work on iOS and Windows - and then not on Android or Windows. 

That the BASIC items are not working is very worrying.  And that learning resources have not been checked for what MAUI is made for - multi-platform development. 

So what should be "shining best of breed examples" of multi-platform code - becomes a **huge advert** for choosing another development platform. 

And then you check Github for progress on fixing bugs - and it is like looking at a oak tree growing. Not much seem to happen at all.  Things might be happening "under the hood" - but far to many important issues are tagged with "wont be solved for next GA"

Sadly - I think for me -  it is convincing me to focus on another platform. I think will take ? 1-2 years before MAUI is "commercial deployment ready". But I hope that is not the case.

**Comment by vanka78bg** [comment]
.NET deserves a good desktop/mobile story. Without it, the .NET framework would be demoted to yet another web framework. And there is plenty of competition on that front already. There would be little incentive to use .NET for anything, as developers could not reuse their codebase between different types of projects anyway. I am surprised that no one from Microsoft is realizing this. Not investing enough resources in the desktop/mobile app development story is hurting the entire .NET ecosystem down the road.
